src/request/sign-message/SignMessage.js

```diff
@@ -0,0 +1,137 @@
+/* global Nimiq */
+/* global KeyStore */
+/* global Identicon */
+/* global PassphraseBox */
+/* global Utf8Tools */
+/* global KeyStore */
+
+class SignMessage {
+    /**
+     * @param {HTMLDivElement} $page
+     * @param {KeyguardRequest.ParsedSignMessageRequest} request
+     * @param {Function} resolve
+     * @param {Function} reject
+     */
+    constructor($page, request, resolve, reject) {
+        /** @type {HTMLDivElement} */
+        const $signerIdenticon = ($page.querySelector('#signer-identicon'));
+
+        /** @type {HTMLDivElement} */
+        const $signerLabel = ($page.querySelector('#signer-label'));
+
+        /** @type {HTMLDivElement} */
+        const $signerAddress = ($page.querySelector('#signer-address'));
+
+        /** @type {HTMLInputElement} */
+        const $message = ($page.querySelector('#message'));
+
+        // Set signing account.
+        const signerAddress = request.signer.toUserFriendlyAddress();
+        new Identicon(signerAddress, $signerIdenticon); // eslint-disable-line no-new
+        $signerAddress.textContent = signerAddress;
+        if (request.signerLabel) {
+            $signerLabel.textContent = request.signerLabel;
+            $signerLabel.classList.remove('display-none');
+        }
+
+        // Set message.
+        if (Utf8Tools.isValidUtf8(request.message)) {
+            $message.value = Utf8Tools.utf8ByteArrayToString(request.message);
+        } else {
+            $message.value = Nimiq.BufferUtils.toHex(request.message);
+        }
+
+        // Set up passphrase box.
+        /** @type {HTMLFormElement} */
+        const $passphraseBox = (document.querySelector('#passphrase-box'));
+        this._passphraseBox = new PassphraseBox($passphraseBox, {
+            hideInput: !request.keyInfo.encrypted,
+            buttonI18nTag: 'passphrasebox-sign-msg',
+            minLength: request.keyInfo.hasPin ? 6 : undefined,
+        });
+
+        this._passphraseBox.on(
+            PassphraseBox.Events.SUBMIT,
+            passphrase => this._onConfirm(request, resolve, reject, passphrase),
+        );
+
+        // This event cannot throw a 'CANCEL' error like in other requests,
+        // because for sign-message we need to go back to the SignMessageOverview
+        // in the Accounts Manager and not return directly to the caller.
+        this._passphraseBox.on(PassphraseBox.Events.CANCEL, () => window.history.back());
+
+        /** @type {HTMLElement} */
+        const $appName = (document.querySelector('#app-name'));
+        $appName.textContent = request.appName;
+        /** @type {HTMLButtonElement} */
+        const $cancelLink = ($appName.parentNode);
+        $cancelLink.classList.remove('display-none');
+        $cancelLink.addEventListener('click', () => reject(new Error('CANCEL')));
```

Comment:
I suggest moving the lines concerning the cancel link into the api's `onRequest`. It is not for this specific page but for this specific request, and in case the handler gets re-used alongside another this code would run multiple times.

However we have multiple styles of the structure anyways. So this is really about how we want to do this in general. Maybe have a discussion about it with the rest.